src/vs/workbench/api/browser/mainThreadChat.ts

```diff
@@ -54,7 +54,9 @@ export class MainThreadChat extends Disposable implements MainThreadChatShape {
 	}
 
 	$transferChatSession(sessionId: number, toWorkspace: UriComponents): void {
-		this._chatService.transferChatSession(sessionId, URI.revive(toWorkspace));
+		const widget = this._chatWidgetService.lastFocusedWidget;
```

Comment:
You'll want to find the widget that has the right sessionId, you could add a method to the ChatWidgetService for this